### PR TITLE
Correctly build Mimir and Helm chart docs with link checking

### PR DESCRIPTION
**Pull request by jdbaldry**
This fixes links such as https://grafana.com/docs/helm-charts/mimir-distributed/latest/run-production-environment-with-helm/monitor-system-health/#monitor-the-health-of-your-system:~:text=Collect%20metrics%20and%20logs%20without%20the%20Helm%20chart which come from using the information architecture in `main` rather than that of the version referenced by the Helm chart (`release-2.9`).

- Replace all Mimir docs links with `docs/reference` shortcode
  Supports compile time link checking but requires specific build steps that still need to be properly automated.
  At the moment, the `README.md` explains the steps.
  For more information about the shortcode, refer to https://grafana.com/docs/writers-toolkit/write/shortcodes/#docsreference-shortcode.
- Fix links broken due to referencing wrong information architecture.

- Attempt to correctly build documentation in CI.
   
Closes https://github.com/grafana/mimir/issues/5715

**Notes to reviewers:**

1. Please ensure that you can run `make docs` locally and that you see no build errors.
1. Please ensure that you understand the additional magic in `docs/Makefile` and `docs/variables.mk`.
  If not, let me know how I can help with that. Perhaps you want a extra information in the `docs/README.md`?

### CODE REVIEW

File: docs/sources/helm-charts/mimir-distributed/_index.md
Comment (osg-grafana) [inline docs/sources/helm-charts/mimir-distributed/_index.md:17]: Why only Mimir here?

Comment (jdbaldry) [reply docs/sources/helm-charts/mimir-distributed/_index.md:17]: Good question! I should have explained this in the PR description.

`docs/reference` does compile time link checking which means you have to have the documentation available which would be difficult for external contributors.

I think the future, it would be nice to update CI to do the full build and locally suppress the errors that relate to private content by default but expose an option for maintainers to do a full build since they can actually download the repository.

Comment (jdbaldry) [reply docs/sources/helm-charts/mimir-distributed/_index.md:17]: Probably best for a future PR, this one is already complicated enough haha.

File: docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-from-cortex.md
Comment (osg-grafana) [inline docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-from-cortex.md:11]: Nice catch!

Comment (osg-grafana) [inline docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-from-cortex.md:11]: Why not a relref here?

Comment (jdbaldry) [reply docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-from-cortex.md:11]: It does no harm to wrap it in a `relref` but it also takes extra key presses and doesn't give you any additional checks. `relref` only checks link destinations and not anchors.

**Comment by jdbaldry** [comment]
I'll fix the CI error as part of a batch of follow up commits that I'm sure will be needed.

File: docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-from-cortex.md
Comment (osg-grafana) [inline docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-from-cortex.md:264]:
```suggestion
[mimirtool_rules]: "/ -> /docs/mimir/<MIMIR DOCS VERSION>/manage/tools/mimirtool#rules"
[`mimirtool config convert`]: "/ -> /docs/mimir/<MIMIR DOCS VERSION>/manage/tools/mimirtool#convert"
[convert]: "/ -> /docs/mimir/<MIMIR DOCS VERSION>/manage/tools/mimirtool#convert"
[Additional resources metrics]: "/ -> /docs/mimir/<MIMIR DOCS VERSION>/manage/monitor-grafana-mimir/requirements#additional-resources-metrics"
[monitoring mixin dashboards]: "/ -> /docs/mimir/<MIMIR DOCS VERSION>/manage/monitor-grafana-mimir/dashboards"
```

File: docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-from-single-zone-with-helm.md
Comment (osg-grafana) [inline docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-from-single-zone-with-helm.md:12]:
```suggestion
This document explains how to migrate stateful components from single zone to [zone-aware replication] with Helm. The three components in question are the [alertmanager], the [store-gateway], and the [ingester].
```

File: docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/_index.md
Comment (osg-grafana) [inline docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/_index.md:95]:
```suggestion
component, refer to [Ingesters failure and data loss] or [Store-gateway: Blocks sharding and replication].
```

File: docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-from-cortex.md
Comment (jdbaldry) [reply docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-from-cortex.md:264]: This suggestion is not for the current release of the Mimir documentation. There is no "Manage" section in the `v2.9.x` documentation (https://grafana.com/docs/mimir/v2.9.x/).

File: docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/_index.md
Comment (osg-grafana) [inline docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/_index.md:352]:
```suggestion
[Planning Grafana Mimir capacity]: "/ -> /docs/mimir/<MIMIR DOCS VERSION>/manage/run-production-environment/planning-capacity"
[Ingesters failure and data loss]: "/ -> /docs/mimir/<MIMIR DOCS VERSION>/references/architecture/components/ingester#ingesters-failure-and-data-loss"
[Collecting metrics and logs from Grafana Mimir]: "/ -> /docs/mimir/<MIMIR DOCS VERSION>/manage/monitor-grafana-mimir/collecting-metrics-and-logs"
[Store-gateway: Blocks sharding and replication]: "/ -> /docs/mimir/<MIMIR DOCS VERSION>/references/architecture/components/store-gateway#blocks-sharding-and-replication"
[replication across availability zones]: "/ -> /docs/mimir/<MIMIR DOCS VERSION>/configure/configure-zone-aware-replication"
[Configure Grafana Mimir object storage backend]: "/ -> /docs/mimir/<MIMIR DOCS VERSION>/configure/configure-object-storage-backend"
[Installing Grafana Mimir dashboards and alerts]: "/ -> /docs/mimir/<MIMIR DOCS VERSION>/manage/monitor-grafana-mimir/installing-dashboards-and-alerts"
```

File: docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/monitor-system-health.md
Comment (osg-grafana) [inline docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/monitor-system-health.md:144]:
```suggestion
[Installing Grafana Mimir dashboards and alerts]: "/ -> /docs/mimir/<MIMIR DOCS VERSION>/manage/monitor-grafana-mimir/installing-dashboards-and-alerts"
[Collect metrics and logs without the Helm chart]: "/ -> /docs/mimir/<MIMIR DOCS VERSION>/manage/monitor-grafana-mimir/collecting-metrics-and-logs#collect-metrics-and-logs-without-the-helm-chart"
```

**Review by osg-grafana — CHANGES_REQUESTED** [review]
There are some outdated references in here.

File: docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/_index.md
Comment (jdbaldry) [reply docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/_index.md:352]: This suggestion is not correct for the release of the Mimir documentation targeted by the Helm chart documentation. There is no "Manage" section in the `v2.9.x` documentation (https://grafana.com/docs/mimir/v2.9.x/).

File: docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/monitor-system-health.md
Comment (jdbaldry) [reply docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/monitor-system-health.md:144]: This suggestion is not correct for the release of the Mimir documentation targeted by the Helm chart documentation. There is no "Manage" section in the `v2.9.x` documentation (https://grafana.com/docs/mimir/v2.9.x/).

**Comment by jdbaldry** [comment]
The references are correct, Hugo link checking confirms it.

File: docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-from-cortex.md
Comment (osg-grafana) [reply docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-from-cortex.md:264]: Ah, ok. I think we should add a milestone to this PR.

File: docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-from-cortex.md
Comment (osg-grafana) [reply docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-from-cortex.md:11]: TIL :)

File: docs/sources/helm-charts/mimir-distributed/_index.md
Comment (osg-grafana) [reply docs/sources/helm-charts/mimir-distributed/_index.md:17]: Can you open a new issue stemming from your comment because it is confusing to only update one and not the other without context. I indeed like your idea about this.

File: docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/monitor-system-health.md
Comment (osg-grafana) [reply docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/monitor-system-health.md:144]: This PR looks like it is opened against `main`.

Comment (osg-grafana) [reply docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/monitor-system-health.md:144]: Oh! You are handing the version with a parameter instead of a CI/CD label. That threw me off.

Comment (jdbaldry) [reply docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/monitor-system-health.md:144]: I'll need someone more familiar with the Mimir Helm chart release process to ensure that this is targeting the correct branch and also how we get the updates where they need to go to update the Mimir Helm chart latest documentation.

Presently that documentation has broken links (like [the https://grafana.com/docs/helm-charts/mimir-distributed/latest/run-production-environment-with-helm/monitor-system-health/#monitor-the-health-of-your-system](https://grafana.com/docs/helm-charts/mimir-distributed/latest/run-production-environment-with-helm/monitor-system-health/#monitor-the-health-of-your-system:~:text=Collect%20metrics%20and%20logs%20without%20the%20Helm%20chart) that work correctly in this branch.

File: docs/sources/helm-charts/mimir-distributed/_index.md
Comment (jdbaldry) [reply docs/sources/helm-charts/mimir-distributed/_index.md:17]: https://github.com/grafana/mimir/issues/5717 :tada:

File: docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/monitor-system-health.md
Comment (osg-grafana) [reply docs/sources/helm-charts/mimir-distributed/run-production-environment-with-helm/monitor-system-health.md:144]: I have added Krajo because he has context about the release process for Helm chart documentation and was the one to parameterize this initially.

**Comment by krajorama** [comment]
Discussed IRL: 
* priority if to have the correct check going so we can at least fix things by hand
* keep version info in helm chart docs _index.md front matter
* determine release branch name from frontmatter
* what to do with main? 
   * solve separately , we're not sure which direction to go , whether to have "latest" or a fixed "version" in the front matter in main

**Comment by jdbaldry** [comment]
I haven't gotten around to making the changes to the `docs/reference` variable substitution syntax. In order to not block this PR on those changes, I will make sure any future changes are backwards compatible.

**Review by osg-grafana — APPROVED** [review]
Unblocking due to subsequent conversation between Jack and Krajo.

**Comment by osg-grafana** [comment]
>   * solve separately , we're not sure which direction to go , whether to have "latest" or a fixed "version" in the front matter in main

Sounds like this comment should be in a new issue. Want to create one, or did you have something else in mind?

**Comment by jdbaldry** [comment]
I've updated the PR description to accurately request what kind of review I need from present and any future reviewers.

**Comment by osg-grafana** [comment]
@jdbaldry, can you rerun the necessary build test for this to pass?

**Comment by jdbaldry** [comment]
Just retriggered it by browsing to https://cla-assistant.io/check/grafana/mimir?pullRequest=5716. In case it gets stuck on other PRs, you just need to update the argument to the `pullRequest` query parameter to match the PR that you want to get unstuck.

File: docs/Makefile
Comment (krajorama) [inline docs/Makefile:11]: According to the comment in `make-docs` REPOS_PATH can be ":" separated list. So I'd use GIT_ROOT/.. instead and add that to REPOS_PATH or something like that.

File: docs/sources/helm-charts/mimir-distributed/_index.md
Comment (krajorama) [inline docs/sources/helm-charts/mimir-distributed/_index.md:11]: I think we should rename this to MIMIR_DOCS_VERSION and have the same spelling everywhere for easy understanding and grep.

File: docs/sources/helm-charts/mimir-distributed/_index.md
Comment (krajorama) [inline docs/sources/helm-charts/mimir-distributed/_index.md:12]: Let's rename to GEM_DOCS_VERSION if we rename the mimir version - needs update in operations/helm/charts/mimir-distributed/RELEASE.md as well

File: docs/sources/helm-charts/mimir-distributed/_index.md
Comment (krajorama) [inline docs/sources/helm-charts/mimir-distributed/_index.md:24]: Let's rename to MIMIR_DOCS_VERSION for consistency, easy understanding and grep.

**Review by krajorama — CHANGES_REQUESTED** [review]

**Comment by osg-grafana** [comment]
Please use `MIMIR-DOCS-VERSION` instead of `MIMIR_DOCS_VERSION` because it is the convention that we use in user-visible docs.

File: docs/Makefile
Comment (osg-grafana) [reply docs/Makefile:11]: @jdbaldry, what is the next action here?

**Comment by krajorama** [comment]
> Please use `MIMIR-DOCS-VERSION` instead of `MIMIR_DOCS_VERSION` because it is the convention that we use in user-visible docs. Please do not feel blocked by this feedback.

This is not user visible variable, this is used to generate the docs themselves. And since we use it in templates and bash scripts, CI pipelines ; it is the syntax of those languages that apply, which all state that identifiers use "_". So I'd keep the underscore.

File: docs/Makefile
Comment (jdbaldry) [reply docs/Makefile:11]: Thanks for catching, resolved in [e156af8](https://github.com/grafana/mimir/pull/5716/commits/e156af81dcd93284b57c6a50e9fb661792acdbe7)

File: .github/workflows/test-build-deploy.yml
Comment (krajorama) [inline .github/workflows/test-build-deploy.yml:281]:
```suggestion
            -v "${PWD}/docs/sources/mimir:/hugo/content/docs/mimir/latest" \
```

File: docs/variables.mk
Comment (krajorama) [inline docs/variables.mk:11]: This is missing "mimir", so you don't get the current mimir documentation on main rendered. Unfortunately if you do add mimir, it's rendered wrong. Latest mimir link will show the versioned documentation and there's errors like 
```
WARN  File "docs/mimir/latest/manage" not found, skipping.
WARN  File "docs/mimir/latest/query" not found, skipping.
WARN  File "docs/mimir/latest/send" not found, skipping.
WARN  File "docs/mimir/latest/set-up" not found, skipping.
WARN  File "docs/mimir/latest/visualize" not found, skipping.
```

**Review by krajorama — APPROVED** [review]
LGTM, tested build locally

**Pull request merged**
